boards/arm/stm32h7/nucleo-h743zi2/src/stm32_bringup.c

```diff
@@ -96,6 +96,21 @@ int stm32_bringup(void)
     }
 #endif
 
+#ifdef CONFIG_STM32_IWDG
+  /* Initialize the watchdog timer */
+
+  stm32_iwdginitialize("/dev/watchdog0", STM32_LSI_FREQUENCY);
```

Comment:
@slorquet you can try the built-in watchdog monitor instead:
https://github.com/apache/nuttx/blob/master/drivers/timers/Kconfig#L395-L464

Reply:
Why do so? I need to control which task resets the watchdog.

Reply:
Followup: I did that and that works.

However, it ENTIRELY masks why I got stack corruption at context switch when trying to create a kthread like I wanted to contribute. This is still worrying.